server/etcdserver/v3_server.go

```diff
@@ -258,6 +258,9 @@ func (s *EtcdServer) Compact(ctx context.Context, r *pb.CompactionRequest) (*pb.
 }
 
 func (s *EtcdServer) LeaseGrant(ctx context.Context, r *pb.LeaseGrantRequest) (*pb.LeaseGrantResponse, error) {
+	if r.ID < int64(lease.NoLease) {
```

Comment:
There are other API methods accepting Lease.ID, should add this validation also to other methods?

Reply:
Yeah, theoretically other API should return return `LeaseNotFound` if leaseID is negative. But it's good to have the validation in place.